app/src/org/commcare/views/media/CommCareMediaController.java

```diff
@@ -50,4 +73,49 @@ public void hide() {
         ViewGroup parent = (ViewGroup) this.getParent();
         parent.setVisibility(View.GONE);
     }
+
+    @Override
+    public void setAnchorView(View view) {
+        super.setAnchorView(view);
+
+        int videoViewId = fullscreenMode ? R.id.fullscreen_video_view : R.id.inline_video_view;
+        CommCareVideoView videoView = view.findViewById(videoViewId);
+
+        if (videoView != null) {
+            addFullscreenButton(videoView);
+        }
+    }
+
+    private void addFullscreenButton(CommCareVideoView videoView) {
+        if (fullscreenBtn == null) {
+            fullscreenBtn = new ImageButton(getContext(), null, R.style.MediaButton);
+            fullscreenBtn.setId(AndroidUtil.generateViewId());
+            if (fullscreenMode) {
+                fullscreenBtn.setImageResource(R.drawable.ic_media_exit_fullscreen);
+            } else {
+                fullscreenBtn.setImageResource(R.drawable.ic_media_fullscreen);
+            }
+            fullscreenBtn.setOnClickListener(view1 -> {
+                // if in fullscreen mode, we exit
+                if (fullscreenMode) {
+                    Intent i = new Intent();
+                    i.putExtra(CommCareMediaController.INLINE_VIDEO_TIME_POSITION, videoView.getCurrentPosition());
+                    ((AppCompatActivity)getContext()).setResult(Activity.RESULT_OK, i);
+                    ((AppCompatActivity)getContext()).finish();
+                } else {
+                    Intent intent = new Intent(getContext(), FullscreenVideoViewActivity.class);
+                    intent.setData(FileUtil.getUriForExternalFile(getContext(),
+                            new File(videoView.getVideoPath())));
```

Comment:
we should explicitly validate videoView.getVideoPath() is not null and crash with an exception otherwise.

Reply:
The [validation of the of the video path](https://github.com/dimagi/commcare-android/blob/123590ee3a6f6c3d3a29fbf89c13c92f9a9544de/app/src/org/commcare/views/media/MediaLayout.java#L306) happens before the instantiation of the CommCareMediaController, so this branch would only be executed if the file exists.